PNA.mdk

```diff
@@ -902,24 +904,39 @@ A PNA implementation need not support both `pna_implementation` and
 ## Tables with add-on-miss capability {#sec-add-on-miss}
 
 PNA defines the `add_on_miss` table property.  If the value of this
-property is `true` for a table, the P4 developer is allowed to define
-a default action for the table that calls the `add_entry` extern
-function.  `add_entry` adds a new entry to the table whose default
-action calls the `add_entry` function.  The new entry will have the
-same key that was just looked up.
-
-The control plane API is still allowed to add, modify, and delete
-entries of such a table, but any entries added via the `add_entry`
-function do not require the control plane software to be involved in
-any way.  It is expected that PNA implementations will be able to
-sustain `add_entry` calls at a large fraction of their line rate, but
-it need not be at the same packet rate supported for processing
-packets that do not call `add_entry`.  The new table entry will be
-matchable when the next packet is processed that applies this table.
+property is `true` for a table `t`, the P4 developer is allowed to
+define a default action for `t` that calls the `add_entry` extern
+function.
+
+When `t.apply()` is invoked, `t`'s lookup key is constructed, and the
+entries of the table are searched.  If there is no match, i.e. the
+lookup results in a miss, `t`'s default action is executed.  So far,
+this is all standard behavior as defined in the P4~16~ language
+specification.
+
+If `t`'s default action makes a call to `add_entry`, it causes a new
+entry to be added to the table with the same key that was just looked
+up and resulted in a miss, and the action name and action parameters
+specified by the parameters of the call to the `add_entry` extern
+function.  Thus, future packets that invoke `t.apply()` with the same
+lookup key will get a match and invoke the specified action (until and
```

Comment:
Does this not assume in-order processing of packets across multiple flows? Should we not state this explicitly?

Reply:
I agree that there is a potentially subtle interaction here between pipelines that can have packets "pass each other up" in the middle of the pipeline, versus those that do not, and it would be good to address this issue somehow in the PNA spec.  I will create a separate issue to track this, since I doubt I will quickly find an acceptable description, and that property of a packet processing pipeline probably affects other features besides this one.

Reply:
Created this issue: https://github.com/p4lang/pna/issues/96  Feel free to add comments on the topic there, but I hope it is reasonable to allow this PR to be merged before that issue is fully resolved.

Reply:
Agree, we should merge this.